_specifications/psp/0.1/plugin/httpRequest.md

```diff
@@ -0,0 +1,88 @@
+#### <a href="#httpRequest" name="httpRequest" class="anchor">Make an Http Request</a>
+
+> *Since version 0.1.0*
+
+The `HttpRequest` request is sent from the server to the client when it needs to access a resource over the network.
+
+*Client Capability*:
+
+* property name (optional): `psp.httpRequest`
+
+*Server Capability*:
+
+* property name (optional): `psp.httpRequest`
+
+*Request*:
+
+* method: `psp/httpRequest`
+* params: `HttpRequestParams` defined as follows:
+
+<div class="anchorHolder"><a href="#httpRequestParams" name="HttpRequestParams" class="linkableAnchor"></a></div>
+
+```typescript
+export interface HttpRequestParams {
+    /**
+     * The HTTP method
+    */
+    method: "GET" | "POST" | "PUT" | "DELETE";
+    /**
+     * The URL to send the request to.
+    */
+    url: URI;
+    /**
+     * The output file where response data should be written.
+     * If "response", data is sent in the response of the request.
+    */
+    output: URI | "response";
+    /**
+     * List of all headers of the response.
+     * Strings should be of format NAME: VALUE where NAME is the header name, and VALUE is the header value
+     * Headers disallowed: Content-Length
+    */
+    headers: string[];
+    /**
+     * Follow redirects.
+     * Integer is the number of max hops.
+     * True allows unlimited hops
+     * False allows no hops
+     * 'default' means the http client gets to decide.
+    */
+    redirects: boolean | integer | 'default';
```

Comment:
Probably `redirects?: boolean | integer;` would be more natural. If left undefined, then the client-editor decides. As well, most requestees do not need to think about this.

Reply:
Also, should we make 'unlimited hops' weaker? A client-editor (or their HTTP impl) may want to throttle it in some way, and thus not completely honor the 'unlimited hops'. I'd go for "True allows unlimited hops up to the limit of the client's maximum"